Slash/Utility/Comments/Comments.pm

```diff
@@ -805,120 +562,13 @@ sub _print_cchp {
-			last unless $pid;
-			$max_depth_allowed++;
-			$tmpcid = $pid;
-		}
-	}
-
-	# The below algorithm assumes that comments are inserted into
-	# the database with cid's that increase chronologically, in order
-	# words that for any two comments where A's cid > B's cid, that
-	# A's timestamp > B's timestamp also.
-
-	for my $x (sort { $a <=> $b } keys %$comments) {
-		next if $x == 0; # exclude the fake "cid 0" comment
-
-		my $pid = $comments->{$x}{pid} || 0;
-		my $reparent = 0;
-
-		# First, if this comment is above the user's desired threshold
-		# (and thus will likely be shown), but its parent is below
-		# the desired threshold (and thus will not likely be shown),
-		# bounce it up the chain until we can reparent it to a comment
-		# that IS being shown.  Effectively we pretend the invisible
-		# comments between this comment and its (great-etc.) grandparent
-		# do not exist.
-		#
-		# But, if all its (great-etc.) grandparents are either invisible
-		# or chronologically precede the root comment, don't reparent it
-		# at all.
-		# XXX either $comments->{$x}{points} or $threshold is sometimes undefined here, not sure which or why
-		if ($user->{reparent} && $comments->{$x}{points} >= $threshold) {
-			my $tmppid = $pid;
-			while ($tmppid
-				&& $comments->{$tmppid} && defined($comments->{$tmppid}{points})
-				&& $comments->{$tmppid}{points} < $threshold) {
-				$tmppid = $comments->{$tmppid}{pid} || 0;
-				$reparent = 1;
-			}
-
-			if ($reparent && $tmppid >= $root_cid_or_pid) {
-				$pid = $tmppid;
-			} else {
-				$reparent = 0;
-			}
-		}
-
-		# Second, if the above did not find a suitable (great)grandparent,
-		# we try a second method of collapsing to a (great)grandparent:
-		# check whether the depth of this comment is too great to show
-		# nested so deeply, and if so, ratchet it back.  Note that since
-		# we are iterating through %$comments in cid order, the parents of
-		# this comment will already have gone through this code and thus
-		# already should have their {depth}s set.  (At least that's the
-		# theory, I'm not sure that part really works.)
-		if ($max_depth_allowed && !$reparent) {
-			# set depth of this comment based on parent's depth
-			$comments->{$x}{depth} = ($pid ? ($comments->{$pid}{depth} ||= 0) : 0) + 1;
-
-			# go back each pid until we find one with depth less than $max_depth_allowed
-			while ($pid && defined($comments->{$pid})
-				&& ($comments->{$pid}{depth} ||= 0) >= $max_depth_allowed) {
-				$pid = $comments->{$pid}{pid};
-				$reparent = 1;
-			}
-		}
-
-		if ($reparent) {
-			# remove child from old parent
-			if ($pid >= $root_cid_or_pid) {
-				@{$comments->{$comments->{$x}{pid}}{kids}} =
-					grep { $_ != $x }
-					@{$comments->{$comments->{$x}{pid}}{kids}}
-			}
-
-			# add child to new parent
-			$comments->{$x}{pid} = $pid;
-			push @{$comments->{$pid}{kids}}, $x;
-		}
-	}
-}
-
 # I wonder if much of this logic should be moved out to the theme.
```

Comment:
should we fix these comments.  as we moved some of this shit back into the templates.